error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/OptionalRules.java

```diff
@@ -422,12 +437,22 @@ Optional<? extends T> after(Optional<S> optional, Function<? super S, ? extends
     }
   }
 
+  /** Prefer {@link Optional#stream()} over more contrived alternatives. */
+  static final class OptionalStream<T> {
+    @BeforeTemplate
+    Stream<T> before(Optional<T> optional) {
+      return Refaster.anyOf(
+          optional.map(Stream::of).orElse(Stream.empty()),
+          optional.map(Stream::of).orElseGet(Stream::empty));
+    }
+
+    @AfterTemplate
+    Stream<T> after(Optional<T> optional) {
+      return optional.stream();
+    }
+  }
+
   // XXX: Add a rule for:
   // `optional.flatMap(x -> pred(x) ? Optional.empty() : Optional.of(x))` and variants.
   // (Maybe canonicalize the inner expression. Maybe we rewrite already.)
```

Comment:
```suggestion
  // (Maybe canonicalize the inner expression. Maybe we rewrite it already.)
```
A tiny improvement IMO.